Can't build with the new Apple SDK

My OSX machine was updated to the new SDK (possibly with the latest OS update or around that time) and it looks like llvm headers that come with it now have the enum for things like `UNW_AARCH64_X19`, so the workaround for not having those constants https://github.com/dotnet/runtime/blob/4ece8f0e73c72271b3e4afa95bfeb7d28ec50869/src/coreclr/pal/src/exception/seh-unwind.cpp#L58

now causes a bunch of build failures like:

```
  In file included from /Users/vs/dotnet02/runtime/src/coreclr/pal/src/exception/seh.cpp:408:
  /Users/vs/dotnet02/runtime/src/coreclr/pal/src/exception/seh-unwind.cpp:63:5: error: redefinition of enumerator 'UNW_AARCH64_X20'
      UNW_AARCH64_X20 = UNW_ARM64_X20,
      ^
  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/include/libunwind.h:519:3: note: previous definition is here
    UNW_AARCH64_X20 = 20,
    ^
  In file included from /Users/vs/dotnet02/runtime/src/coreclr/pal/src/exception/seh.cpp:408:
  /Users/vs/dotnet02/runtime/src/coreclr/pal/src/exception/seh-unwind.cpp:64:5: error: redefinition of enumerator 'UNW_AARCH64_X21'
      UNW_AARCH64_X21 = UNW_ARM64_X21,
      ^
  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/include/libunwind.h:520:3: note: previous definition is here
    UNW_AARCH64_X21 = 21,
    ^
```

Not sure how to deal with that. I assume we'd want to be able to build with older SDKs as well, at least for some time. 
Perhaps the SDK version can be known at build time and the workaround can be conditionally applied?

--- a/src/coreclr/pal/src/config.h.in
+++ b/src/coreclr/pal/src/config.h.in
@@ -67,6 +67,7 @@
 #cmakedefine01 HAVE_SCHED_SETAFFINITY
 #cmakedefine HAVE_UNW_GET_SAVE_LOC
 #cmakedefine HAVE_UNW_GET_ACCESSORS
+#cmakedefine HAVE_UNW_AARCH64_X19
 #cmakedefine01 HAVE_XSWDEV
 #cmakedefine01 HAVE_XSW_USAGE
 #cmakedefine01 HAVE_PUBLIC_XSTATE_STRUCT

--- a/src/coreclr/pal/src/configure.cmake
+++ b/src/coreclr/pal/src/configure.cmake
@@ -1032,6 +1032,15 @@ int main(int argc, char **argv)
 check_symbol_exists(unw_get_save_loc libunwind.h HAVE_UNW_GET_SAVE_LOC)
 check_symbol_exists(unw_get_accessors libunwind.h HAVE_UNW_GET_ACCESSORS)
 
+check_cxx_source_compiles("
+#include <libunwind.h>
+
+int main(int argc, char **argv)
+{
+    int flag = (int)UNW_AARCH64_X19;
+    return 0;
+}" HAVE_UNW_AARCH64_X19)
+
 if(NOT CLR_CMAKE_USE_SYSTEM_LIBUNWIND)
   list(REMOVE_AT CMAKE_REQUIRED_INCLUDES 0 1)
 endif()

--- a/src/coreclr/pal/src/exception/seh-unwind.cpp
+++ b/src/coreclr/pal/src/exception/seh-unwind.cpp
@@ -54,7 +54,7 @@ Abstract:
 
 #endif // HOST_UNIX
 
-#if defined(TARGET_OSX) && defined(HOST_ARM64)
+#if defined(TARGET_OSX) && defined(HOST_ARM64) && !defined(HAVE_UNW_AARCH64_X19)
 // MacOS uses ARM64 instead of AARCH64 to describe these registers
 // Create aliases to reuse more code
 enum

